packages/app/src/systems/Core/components/CoinInput.tsx

```diff
@@ -77,3 +62,36 @@ export const CoinInput = forwardRef<HTMLInputElement, CoinInputProps>(
     );
   }
 );
+
+function useDisplayedCoins(
```

Comment:
Instead of creating a new hook, I think this hook changes should go inside already existing `useCoinInput` hook, as which is the responsible for managing CoinInput.

Reply:
The proposal makes sense, but unfortunately `useCoinInput` is not being used in any functionality related to swapping (the swap widget has its own `useSwapCoinInput` hook which doesn't reference `useCoinInput`), so there'd have to be some additional refactoring going on that's IMO out of scope for these two issues. This would I think better be done in a separate issue that's related to #306. I'd be happy to do it, but I believe it's worth a separate issue.

Reply:
yeah it would need to be implemented both in `useCoinInput` and `useSwapCoinInput`, we can refact it in other issue.

1- For now then we should move this hook to `hooks` folder, instead of living inside `CoinInput` component. Suggestion for name is `useCoinInputDisplayValue`
2- We have a broken test related to coin input value in other page, which may be a side effect. Do you know why the test is broke? We cannot merge it with broken tests so we may either fix it or if it's a false alert skip the test and create a new issue to fix it later

Reply:
1- Done.
2- I fixed it. The hook was using `e.currentTarget.value` whereas the test was firing a change event with a `target` object.